flake.nix

```diff
@@ -2,128 +2,107 @@
-        dream2nix.lib.makeFlakeOutputs {
-          systems = supportedSystems;
-          config.projectRoot = src;
-          source = src;
-          settings = [
-            {
-              subsystemInfo.noDev = false;
-              subsystemInfo.nodejs = 16;
-            }
-          ];
+      d2nFlake = dream2nix.lib.makeFlakeOutputs rec {
+        inherit systems;
+        source = builtins.path {
+          path = ./.;
+          filter = path: _: builtins.baseNameOf path != "package-lock.json";
         };
+        config.projectRoot = source;
+        settings = [{
+          subsystemInfo.noDev = false;
+          subsystemInfo.nodejs = 16;
+        }];
+      };
 
-      artifactsBuilder = system: envVars:
-        let
-          nft-marketplace-frontend = d2nFlakeOutputs.packages.${system}.default.overrideAttrs (old: {
-            preBuild =
-              let
-                seabug-contracts-output-js = seabug-contracts.packages.${system}.seabug-contracts.overrideAttrs (old: {
-                  buildCommand = old.buildCommand + ''
-                    cp output.js $out
-                  '';
-                });
-              in
-              (old.preBuild or "") + ''
-                mkdir seabug-contracts-module
-                cp -r $out/lib/node_modules/nft-marketplace/node_modules/seabug-contracts/* seabug-contracts-module
-                cp ${seabug-contracts-output-js}/output.js seabug-contracts-module
-                rm -rf $out/lib/node_modules/nft-marketplace/node_modules/seabug-contracts
-                cp -r seabug-contracts-module $out/lib/node_modules/nft-marketplace/node_modules/seabug-contracts
-              '';
-          } // envVars);
-        in
-        pkgsFor.${system}.stdenv.mkDerivation {
-          name = "nft-marketplace";
-          dontUnpack = true;
-          dontBuild = true;
-          installPhase = ''
-            cp -r ${nft-marketplace-frontend}/lib/node_modules/nft-marketplace/build $out
-          '';
-        };
+      # TODO: upstream
+      make-seabug-contracts-output-js = system: inputs.seabug-contracts.packages.${system}.seabug-contracts.overrideAttrs (old: {
+        buildCommand = old.buildCommand + ''
+          cp output.js $out
+        '';
+      });
+
+      make-frontend = system: buildEnvVars: d2nFlake.packages.${system}.default.overrideAttrs (old: {
+        preBuild = (old.preBuild or "") + ''
+          mkdir seabug-contracts-module
+          cp -r $out/lib/node_modules/nft-marketplace/node_modules/seabug-contracts/* seabug-contracts-module
+          cp ${make-seabug-contracts-output-js system}/output.js seabug-contracts-module
+          rm -rf $out/lib/node_modules/nft-marketplace/node_modules/seabug-contracts
+          cp -r seabug-contracts-module $out/lib/node_modules/nft-marketplace/node_modules/seabug-contracts
+          # avoid Error: error:0308010C:digital envelope routines::unsupported
+          export NODE_OPTIONS=--openssl-legacy-provider
```

Comment:
Why did you need to add this? I think I've never got this error, what did you change to have it?